### PR TITLE
Fix Time::Span multiply and divide

### LINKED ISSUE #5272: Time::Span#*, Time::Span#/ and Float
Trying to calculate an estimate of remaining time for a task based on elapsed time and percentage done, using the following code:

```.ruby
done = 0.75 # 75% of work done
elapsed = Time::Span.new(0,1,0)
total = elapsed / done
```

I see the following error:

    Error in line 3: instantiating 'Time::Span#/(Float64)'

    in /scr/hallasan/crystal-0.24.0-2/src/time/span.cr:303: no overload matches 'Int64#tdiv' with type Float64
    Overloads are:
     - Int#tdiv(other : Int)

        seconds = to_i.tdiv(number)
                       ^~~~

Alternatively:

```.ruby
done = 0.75 # 75% of work done
elapsed = Time::Span.new(0,1,0)
total = elapsed * (1.0/done)
```

results in

    Error in line 3: instantiating 'Time::Span#*(Float64)'

    in /scr/hallasan/crystal-0.24.0-2/src/time/span.cr:296: no argument named 'seconds'
    Matches are:
     - Time::Span.new(days : Int, hours : Int, minutes : Int, seconds : Int, nanoseconds : Int = 0)
     - Time::Span.new(hours : Int, minutes : Int, seconds : Int)
     - Time::Span.new(*, seconds : Int, nanoseconds : Int)
     - Time::Span.new(*, nanoseconds : Int) (trying this one)

        Span.new(
             ^~~

So, neither `Time::Span#/` nor `Time::Span#*` appear to accept floats. However, the API documentation gives their types as `#*(number : Number)` and `#/(number : Number)`, and `0.75.is_a? Number # => true`.

Am I missing something or is this a bug?

```
$ crystal -v
Crystal 0.23.0+313 [ea4187c57] (2017-10-27)

LLVM: 3.9.1
Default target: x86_64-unknown-linux-gnu
```

### PR DESCRIPTION
**Pull request by petoem**
Fixes #5272

### CODE REVIEW

**Comment by asterite** [comment]
If it fixes something there should be a spec showing what was broken.

**Comment by petoem** [comment]
@asterite The [spec](https://github.com/crystal-lang/crystal/blob/master/spec/std/time/span_spec.cr#L207) only test multiply and divide with `Int`, that's why they never failed.

**Comment by asterite** [comment]
@petoem Could you add tests that show it works with `Float` arguments then?

File: src/time/span.cr [hunk 1]
```diff
@@ -291,15 +291,21 @@ struct Time::Span
...
     )
   end
 
-  def /(number : Number) : Time::Span
+  # Returns a `Time::Span` that is *number* times longer.
+  def *(number : Float) : Time::Span
+    (total_nanoseconds * number).nanoseconds
+  end
+
+  # Return a `Time::Span` that is divided by *number*.
```
Comment (straight-shoota) [inline src/time/span.cr]: `Returns`

Comment (petoem) [reply src/time/span.cr]: Thanks, nice catch.

**Pull request merged**